web/thesauruses/javascript/ECMAScript 2009/classes.json

```diff
@@ -0,0 +1,108 @@
+      "name": "Read-only class"
+    },
+    "static_class": {
+      "not-implemented": true,
+      "name": "Static class"
+    },
+    "inner_class": {
+      "not-implemented": true,
+      "name": "Inner class"
+    },
+    "packages": {
+      "comment": "A package manager like NPM (https://www.npmjs.com/) or Deno (https://deno.land/) can be used to create and manage Javascript packages.",
+      "not-implemented": true,
+      "name": "Packages"
+    },
+    "class_with_generic_type": {
+      "not-implemented": true,
+      "name": "Class with a generic type"
+    },
+    "private_variables": {
+      "code": "var ClassName = (function() {\n var privateField = 0;\n // privateField is accessible in this scope only.\n function ClassName() {\n  // Constructor body\n }\n return ClassName;\n})();",
+      "name": "Defining private variables"
+    },
+    "protected_variables": {
+      "not-implemented": true,
+      "name": "Defining protected variables"
+    },
+    "public_variables": {
+      "code": "function ClassName() {\n // Constructor body\n this.publicField = 0\n}\nClassName.prototype.anotherPublicField = 0;",
+      "name": "Defining public variables"
+    },
+    "static_variables": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.staticField = 0;\n return ClassName;\n})();",
+      "name": "Defining static variables"
+    },
+    "private_functions": {
+      "code": "var ClassName = (function() {\n var privateMethod = function() {\n  // Method body\n }\n // privateMethod is accessible in this scope only.\n function ClassName() {\n  // Constructor body\n }\n return ClassName;\n})();",
+      "name": "Defining private functions"
+    },
+    "protected_functions": {
+      "not-implemented": true,
+      "name": "Defining protected functions"
+    },
+    "public_functions": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.prototype.publicMethod = function() {\n  // Method body\n }\n return ClassName;\n})();",
+      "name": "Defining public functions"
+    },
+    "static_functions": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.staticMethod = function() {\n  // Method body\n }\n return ClassName;\n})();",
+      "name": "Defining static functions"
+    },
+    "extends_class": {
+      "code": "var ParentClassName = function ParentClassName() {\n // Constructor body\n}\nvar ChildClassName = (function() {\n function ChildClassName() {\n  // Call the parent class constructor, should be first line of constructor\n  ParentClassName.call(this);\n  // Constructor body\n }\n // Copy the prototype from the parent class\n ChildClassName.prototype = Object.create(ParentClassName.prototype);\n // Set the constructor to refer to this class\n ChildClassName.prototype.constructor = ChildClassName;\n return ChildClassName;\n})();",
```

Comment:
Code Thesaurus is supposed to be a language reference tool and not an algorithm tool. At this point you're pretty much writing out how to do everything with subclasses, and that's a bit different than a way to have inherited classes which is what this is suggesting. If JS doesn't have a built-in way to do inheritable classes, then I'd just make this "not-implemented".

Reply:
It's done !